src/libcore/char.rs

```diff
@@ -415,12 +415,12 @@ pub struct EscapeUnicode {
 
 #[derive(Clone, Debug)]
 enum EscapeUnicodeState {
-    Backslash,
-    Type,
-    LeftBrace,
-    Value,
-    RightBrace,
     Done,
+    RightBrace,
+    Value,
+    LeftBrace,
+    Type,
+    Backslash,
```

Comment:
Out of curiosity, why reorder these? The previous order makes a little more sense in my mind as we tend to read top-to-bottom